### PR TITLE
Deprecate S2758 in favor of S3923

### LINKED ISSUE #1670: Deprecate S2758 in favor of S3923
### Description

This rule is now deprecated and is replaced by S3923.

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #1670

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AllBranchesShouldNotHaveSameImplementation.cs [hunk 1]
```diff
@@ -100,7 +100,8 @@ protected override void Initialize(SonarAnalysisContext context)
                 {
                     var ternaryStatement = (ConditionalExpressionSyntax)c.Node;
 
-                    if (ternaryStatement.WhenTrue.IsEquivalentTo(ternaryStatement.WhenFalse, false))
+                    if (ternaryStatement.WhenTrue.RemoveParentheses().IsEquivalentTo(
+                            ternaryStatement.WhenFalse.RemoveParentheses(), false))
```
Comment (valhristov) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AllBranchesShouldNotHaveSameImplementation.cs]: Is this related to the current change set?

Comment (Evangelink) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AllBranchesShouldNotHaveSameImplementation.cs]: Yes when copying the test cases from the deprecated rules I have noticed we were not handling parenthesis (on the new rule) while the deprecated one was.
